### ISSUE TITLE
n-fold duals

### ISSUE THREAD

**Issue by calintat** [homotopy-io/homotopy-rs#1004]
Now that Klaudia has an initial version of manifold detection, it's time to think about implementing this into the tool.

For every generator, we should have an option to make it $n$-dualisable rather than invertible. If a generator is $n$-dualisable, the only homotopy moves allowed in dimensions $> n$ are the ones where the singular slice is manifold-like. @jamievicary Is that correct?

**Comment by jamievicary** [comment]
Yes this is broadly correct but let's see if we can make a more precise statement.

For an $n$-dualizable $k$-dimensional generator $G$, consider a $d+1$-dimensional homotopy. If $d+1 \leq k+n$, there are no additional checks to perform. Otherwise we examine the $d$-dimensional singular target slice of the homotopy, and consider the locus of $G$. If it is not a $(d-1)$-dimensional manifold then homotopy construction fails.

If anyone thinks some details here might be wrong it would be good to hear that. Otherwise let's implement it and see if it behaves the way we are expecting.
